spec/features/survey_response_new_survey_spec.rb

```diff
@@ -62,10 +62,13 @@
     survey_response = create(:survey_response, user: User.last)
     survey_response_2 = create(:survey_response, user: User.last, notes: nil)
 
-    expect(page.first("tr")).to have_content(survey_response.created_at)
-    expect(page.first("tr")).to have_content(survey_response.perceived_code_confidence)
-    expect(page.first("tr")).to have_content(survey_response.material_difficulty_rating)
-    expect(page.first("tr")).to have_content(survey_response.notes)
-    expect(page.last("tr")).to have_content(survey_response.notes) # does this work?
+    visit survey_responses_path
+    save_and_open_screenshot
+
+    expect(page.first("tr:nth-of-type(2)")).to have_content(survey_response.created_at.in_time_zone('Eastern Time (US & Canada)').strftime("%b %e, %Y"))
+    expect(page.first("tr:nth-of-type(2)")).to have_content(survey_response.perceived_code_confidence)
+    expect(page.first("tr:nth-of-type(2)")).to have_content(survey_response.material_difficulty_rating)
+    expect(page.first("tr:nth-of-type(2)")).to have_content(survey_response.notes)
+    expect(page.last("tr")).to have_content(survey_response.notes)
```

Comment:
@kopchickm Running this spec doesn't pass because it's pulling in info from the last spec. So if I do a pry and check what `survey_response` and `survey_response_2` are, it is what's in the factory, but then the save and open page shows me that what's there is what has been generated from previous specs in the file. Additionally, if I run the spec in isolation, then the save_and_open_screenshot just shows me a blank index view (headers but no survey entries). Any thoughts or ideas?

Reply:
My first thought is that `User.last` isn't necessarily going to be the same thing every time, if there are multiple users in the system. You could start by making sure you're signed in as the same user that owns the 2 survey responses.

I'm not sure if there's a database cleaning issue here...you could check to see how many survey responses are in the database just to sanity check that there are only 2.

If you run the test in isolation, does it pass? Are the number of survey responses different if you run the test in isolation vs. running the whole file?

Reply:
@kopchickm It is different. When I run the whole file I get 3 surveys that are filled in with 5s and 2s like the previous tests fill out a survey. When I run just that spec in isolation, I get a blank index page with no surveys and just the headers. I messed around with truncation and transaction and even commenting out all the database cleaner code and nothing changes. It’s like the factories aren’t being called for some reason. It’s super weird and has me totally stumped and stuck. I also know that the first tr is the header. There isn’t a way to call the 2nd or last tr and I don’t think nth of a type works but I figured I need to figure out the first issue before moving on to the expectations.